src/main/java/org/opensearch/neuralsearch/sparse/cache/ForwardIndexCacheItem.java

```diff
@@ -63,19 +65,30 @@ public SparseVector read(int docId) throws IOException {
             if (docId >= sparseVectors.length()) {
                 return null;
             }
-            return sparseVectors.get(docId);
+            SparseVector vector = sparseVectors.get(docId);
+            if (vector != null) {
+                // Record access to update LRU status
+                LruDocumentCache.getInstance().updateAccess(cacheKey, docId);
+            }
+            return vector;
         }
     }
 
-    private class CacheSparseVectorWriter implements SparseVectorWriter {
+    private class CacheSparseVectorWriter implements CacheableSparseVectorWriter {
+
         private final Consumer<Long> circuitBreakerTriggerHandler;
 
-        private CacheSparseVectorWriter(Consumer<Long> circuitBreakerTriggerHandler) {
-            this.circuitBreakerTriggerHandler = circuitBreakerTriggerHandler;
+        // Default handler: perform cache eviction when memory limit is reached
+        private CacheSparseVectorWriter() {
```

Comment:
I think the current way of implementation of two getWritter() isn't quite right. It's not that writer is created once and getWriter always return the same instance. I'll try to refactor it later.